### PR TITLE
fix: sync RTC in timed, sync time before fetching packet metadata

### PR DESCRIPTION
**Pull request by smira**
1. When time is set by timed, also sync RTC with system time, otherwise
time might be off after a reboot.

2. Before fetching Packet (Equinix Metal) metadata over `https://`, do
one-time time sync in the background, as if the clock skew is big
enough, TLS certificate will be considered invalid and boot fails.

Signed-off-by: Andrey Smirnov <smirnov.andrey@gmail.com>

### CODE REVIEW

**Review by rsmitty — DISMISSED** [review]

File: internal/app/machined/pkg/runtime/v1alpha1/platform/packet/packet.go [hunk 1]
```diff
@@ -77,6 +79,24 @@ func (p *Packet) Name() string {
...
+			return
+		}
+
+		if err = ntpClient.QueryAndSetTime(ntpCtx); err != nil {
+			if !stderrors.Is(err, context.Canceled) {
+				log.Printf("failed to do one-time time sync: %s", err)
+			}
+		}
+	}()
+
```
Comment (andrewrynhard) [inline internal/app/machined/pkg/runtime/v1alpha1/platform/packet/packet.go:99]: Feels like maybe we can put this in a more generic place? Maybe in the task that calls the platform interface `Configuration` method?

**Review by andrewrynhard — COMMENTED** [review]
LGTM, but just one suggestion.

File: internal/app/machined/pkg/runtime/v1alpha1/platform/packet/packet.go (hunk 1 above)
Comment (smira) [reply internal/app/machined/pkg/runtime/v1alpha1/platform/packet/packet.go:99]: yep, I think it makes perfect sense to do that there.

**Review by andrewrynhard — DISMISSED** [review]

**Pull request merged**